go-controller/pkg/ovn/policy_common.go

```diff
@@ -197,11 +197,19 @@ func (oc *Controller) handlePeerPodSelectorAddUpdate(np *namespacePolicy,
 	addressMap map[string]bool, addressSet string, obj interface{}) {
 
 	pod := obj.(*kapi.Pod)
-	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
-	if err != nil {
+	var ipAddress string
+	if PodWantsNetwork(pod) {
+		podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
+		if err == nil {
+			ipAddress = podAnnotation.IP.IP.String()
+		}
+	} else {
+		// the peer pod could be on host network
+		ipAddress = pod.Status.PodIP
```

Comment:
NetworkPolicy is not expected to work for hostNetwork pods. (eg, what if you had two hostNetwork pods on the same node and you wanted one of them to be allowed and the other not?). So I think the earlier behavior (ignoring hostNetwork pods) is correct.

Reply:
@danwinship  I see. let me revert my changes.